ext/filter/filter.c

```diff
@@ -690,29 +690,13 @@ PHP_FUNCTION(filter_input_array)
 	}
 
 	array_input = php_filter_get_storage(fetch_from);
+
 	if (EG(exception)) {
 		RETURN_THROWS();
 	}
 
 	if (!array_input) {
-		zend_long filter_flags = 0;
-		zval *option;
-		if (op_long) {
```

Comment:
The bug here is that `op_long` was checked instead of `!op_ht`, just like what `filter_input` does.

Reply:
at first sight, I though the same, but let me continue on the next response right below this one.